src/battle_tower.c

```diff
@@ -1324,15 +1324,15 @@ void PutNewBattleTowerRecord(struct EmeraldBattleTowerRecord *newRecordEm)
     struct EmeraldBattleTowerRecord *newRecord = newRecordEm; // Needed to match.
 
     // Find a record slot of the same player and replace it.
-    for (i = 0; i < 5; i++)
+    for (i = 0; i < BATTLE_TOWER_RECORD_COUNT; i++)
     {
         k = 0;
         for (j = 0; j < TRAINER_ID_LENGTH; j++)
         {
             if (gSaveBlock2Ptr->frontier.towerRecords[i].trainerId[j] != newRecord->trainerId[j])
                 break;
         }
-        if (j == 4)
+        if (j == BATTLE_TOWER_RECORD_COUNT - 1)
```

Comment:
`j` is looping over `TRAINER_ID_LENGTH`, not `BATTLE_TOWER_RECORD_COUNT`

Reply:
You’re right, my bad. I thought this was an operation specifically on the last iteration of the loop. I’ll fix that.